File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
#include <unordered_map>
#include <map>
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> nums_map;
        vector<vector<int>> ordered_nums (nums.size() + 1);
        vector<int> result;
        for (auto num : nums) {
            nums_map[num]++;
        }
        for (auto [key, value] : nums_map) {
            ordered_nums[value].push_back(key);
        }
        for (int i = ordered_nums.size() - 1; i >= 0 && k > 0; i--) {
            if (!ordered_nums[i].empty()) {
                for (int num : ordered_nums[i]) {
                    if (k > 0) {
                        result.push_back(num);
                        k--;
                    }
                }
            }
        }
        return result;
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        seen = {}
        result = 0
        l,r = 0,0
        while r < len(s):
            if s[r] in seen:
                l = max(l, seen[s[r]] + 1)
            seen[s[r]] = r
            result = max(result, r - l + 1)
            r += 1
        return result